dScripts/RockHydrantBroken.cpp

```diff
@@ -0,0 +1,47 @@
+#include "RockHydrantBroken.h"
+#include "EntityManager.h"
+#include "GameMessages.h"
+
+void RockHydrantBroken::OnStartup(Entity* self)
+{
+	self->AddTimer("playEffect", 1);
+
+	const auto hydrant = "hydrant" + self->GetVar<std::string>(u"hydrant");
+
+	const auto bouncers = EntityManager::Instance()->GetEntitiesInGroup(hydrant);
+
+	Game::logger->Log("RockHydrantBroken", "Broken Rock Hydrant spawned (%s)\n", hydrant.c_str());
+
+	for (auto* bouncer : bouncers)
+	{
+		self->SetVar<LWOOBJID>(u"bouncer", bouncer->GetObjectID());
+
+
+		GameMessages::SendBouncerActiveStatus(bouncer->GetObjectID(), true, UNASSIGNED_SYSTEM_ADDRESS);
+
+		GameMessages::SendNotifyObject(bouncer->GetObjectID(), self->GetObjectID(), u"enableCollision", UNASSIGNED_SYSTEM_ADDRESS);
+	}
+
+	self->AddTimer("KillBroken", 10);
+}
+
+void RockHydrantBroken::OnTimerDone(Entity* self, std::string timerName)
+{
+	if (timerName == "KillBroken")
+	{
+		auto* bouncer = EntityManager::Instance()->GetEntity(self->GetVar<LWOOBJID>(u"bouncer"));
+
+		if (bouncer != nullptr)
+		{
+			GameMessages::SendBouncerActiveStatus(bouncer->GetObjectID(), false, UNASSIGNED_SYSTEM_ADDRESS);
+
+			GameMessages::SendNotifyObject(bouncer->GetObjectID(), self->GetObjectID(), u"disableCollision", UNASSIGNED_SYSTEM_ADDRESS);
+		}
+
+		self->Kill();
```

Comment:
`self->Kill()` will not show the smashing animation, did you intend to use `self->Smash()`? The script this is based on is old, so it might be worth investigating if it was correct.

Reply:
I think Kill() is correct, since there is no model to smash at that point, just the entity that is making the bouncer, which is invisible